Library/Homebrew/cmd/update.rb

```diff
@@ -57,6 +58,7 @@ def update
     else
       puts "Updated Homebrew from #{master_updater.initial_revision[0,8]} to #{master_updater.current_revision[0,8]}."
       report.dump
+      Descriptions.refresh_cache
```

Comment:
Maybe just update/create the cache on first use and use the last update compared with the cache's date to decide whether it needs to be regenerated based on the update?